setup.py

```diff
@@ -21,6 +21,5 @@
           'evalml = evalml.__main__:cli'
         ]
     },
-    data_files=[('evalml/demos/data', ['evalml/demos/data/fraud_transactions.csv.gz', 'evalml/demos/data/churn.csv']),
```

Comment:
If we delete `fraud_transactions.csv.gz` and `churn.csv` from this list, won't that mean that `load_fraud` and `load_churn` will fail for users who pip install and set `use_local=True`?

Reply:
Yeah, `use_local` would fail unless they manually download the files. Since we're defaulting to downloading from S3, it seemed unnecessary to keep them in the package. They are small though, so keeping them in isn't a big deal if that's the way we want to go. 

Also, if we don't want to include the data files but do want to allow `use_local` for users/make it easier to use, I think there's two other options we could do:
1)  Change `use_local` from a boolean to a path. That way users could download a local copy and specify its location instead of downloading from S3 every time.
2) Add a `save_data` flag that saves the CSV after downloading from S3 to the default data directory so that `use_local` would work in the future.

Reply:
I agree with you in that what we should do should depends on whether users are intended to be able to use the local datasets.

1. If we want them to be able to use the local data, then I think the simplest thing to do is to include `fraud_transactions.csv.gz` and `churn.csv` in the package. I don't think we need to add `save_data` or change `use_local` to be a path but let me know what you think.

2. If we want them to always go through s3, then I think what we can do is remove `use_local` from the `load_data` functions. In our tests, we'll create fixtures that load the data locally. Then we can remove the demo data from our pypi package.

I'm partial to 2 (users always go through s3) because it's consistent with what featuretools does and it lets us run the unit tests without an internet connection.

Let me know what you think!

Reply:
@frances-h Thanks for making the changes! This looks great. I thought about it this morning, and I think we'll need to still include fraud and churn in the pypi package. The reason is that we run our unit tests when we build the conda package for a new release and the conda build happens from the tar.gz on PyPi. 

Once that small change is in, feel free to merge! Thank you 😄 

FYI @dsherry @chukarsten